File: sort-mapper-example-java/src/main/java/kh/farrukh/sortmapperexamplejava/lesson/LessonRepo.java
package kh.farrukh.sortmapperexamplejava.lesson;

import java.util.Optional;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface LessonRepo extends JpaRepository<LessonEntity, Long> {

  Optional<LessonEntity> findByIdAndDeletedAtIsNull(Long id);

  Page<LessonEntity> findAllByDeletedAtIsNull(Pageable pageable);

}

File: sort-mapper-example/src/main/kotlin/kh/farrukh/sortmapperexample/lesson/LessonRepo.kt
package kh.farrukh.sortmapperexample.lesson

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface LessonRepo : JpaRepository<LessonEntity, Long> {

    fun findByIdAndDeletedAtIsNull(id: Long): LessonEntity?

    fun findAllByDeletedAtIsNull(pageable: Pageable): Page<LessonEntity>

}